ert/data/_record.py

```diff
@@ -235,8 +270,7 @@ async def load(self) -> Record:
 
     @abstractmethod
     async def transmit_data(
-        self,
-        data: record_data,
+        self, data: Union[numerical_record_data, blob_record_data]
```

Comment:
This type definition does not hold if we introduce new record types and hence it feels like a bad abstraction to me. I see that it was not typed prior to this PR, perhaps we could keep it that way or let it be `Any`?

Reply:
Correct. Here I ran in the bad interactions of pydantic/inheritance/and mypy. One would like to put the base class `Record` here, but that made mypy freak out in other places, this seemed the least worse approach. Also, this function needs to have separate code paths for each  record type anyway, so the abstraction was not there in the first place. Making the typing explicit clarifies that this function really  only works for these two types, and is not going to work for unknown new types. So, unless we abstract the actual code of that function, I would vote for leaving the typing as is.

Reply:
I see. If nothing else, perhaps we could at least introduce a `record_data = Union[numerical_record_data, blob_record_data]` type above and use that the different places we need to type this? It will ensure consistency, communicate some intent and probably make refactoring later less error prone.

Reply:
That sounds like a good idea, will do that.